kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/vault/KubeVaultTestKmsFacade.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.resources.vault;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Objects;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.kroxylicious.kms.provider.hashicorp.vault.VaultKmsService;
+import io.kroxylicious.kms.service.KmsService;
+import io.kroxylicious.kms.service.TestKekManager;
+import io.kroxylicious.kms.service.TestKmsFacade;
+import io.kroxylicious.kms.service.UnknownAliasException;
+import io.kroxylicious.systemtests.executor.ExecResult;
+import io.kroxylicious.systemtests.installation.vault.Vault;
+import io.kroxylicious.systemtests.k8s.exception.KubeClusterException;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.cmdKubeClient;
+
+public class KubeVaultTestKmsFacade implements TestKmsFacade<VaultKmsService.Config, String, VaultEdek> {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final String VAULT_CMD = "vault";
+    private static final String FORMAT_JSON = "-format=json";
+    private final String namespace;
+    private final String podName;
+    @SuppressWarnings("rawtypes")
+    private final Vault vault;
+
+    public KubeVaultTestKmsFacade(String namespace, String podName) {
+        this.namespace = namespace;
+        this.podName = podName;
+        this.vault = new Vault(namespace);
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return vault.isAvailable();
+    }
+
+    @Override
+    @SuppressWarnings("resource")
+    public void start() {
+        vault.deploy();
+    }
+
+    @Override
+    public void stop() {
+        try {
+            vault.delete();
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public TestKekManager getTestKekManager() {
+        return new VaultTestKekManager();
+    }
+
+    @Override
+    public Class<? extends KmsService<VaultKmsService.Config, String, VaultEdek>> getKmsServiceClass() {
+        return null;
+    }
+
+    @Override
+    public VaultKmsService.Config getKmsServiceConfig() {
+        return new VaultKmsService.Config(URI.create(vault.getBootstrap()), Vault.VAULT_ROOT_TOKEN);
+    }
+
+    private class VaultTestKekManager implements TestKekManager {
+
+        public void generateKek(String alias) {
+            Objects.requireNonNull(alias);
+
+            if (exists(alias)) {
+                throw new AlreadyExistsException(alias);
+            }
+            else {
```

Comment:
Why are we preventing duplicate names going to vault? I'd assume the create call will fail if the name already exists. I think the `TestKekManager` should be the thinest wrapper possible around the actual vault instance and the less logic, and fewer remote calls, we have to maintain the better.

Reply:
Ohh is this just about complying with the contract and throwing the appropriate exception types? 

Then I understand why it looks like this but as I said above we want to keep the number of remote invocations to pods as low as possible for performance reasons. So I think we should be handling the vault error responses rather than preventing the problematic calls

Reply:
I understand your concern, but this code is only intended to be used in the system tests, so from my POV the performance is not as important as it should be in the Kroxylicious code. It is more important not adding possible unexpected errors in the framework than having a better performance.

Reply:
> Ohh is this just about complying with the contract and throwing the appropriate exception types?

This is just a copy/paste from `VaultTestKmsFacade` but maybe not needed here.

Reply:
In fact, I've just tested and using `-f` parameter does not return an error if already exists